### ISSUE TITLE
[3.3.5] Quest 4041 The Videre Elixir is bugged

### ISSUE THREAD

**Issue by mario-64** [TrinityCore/TrinityCore#14774]
The Videre Elixir: [http://www.wowhead.com/quest=4041]
Gregan Brewspewer requests to you to bring him an "Evoroot" [http://www.wowhead.com/item=11242].

To obtain it, you must to find Miblon Snarltooth [http://www.wowhead.com/npc=9467]. The Evoroot is inside the ruins behind him. You use the bait that Gregan sells [http://www.wowhead.com/?item=11141] and put it near Miblon. It is supposed to distract him making the wall behind him disappear so you can walk in and get the Evoroot. This does not happen. When you place the bait he does nothing.

See https://github.com/TrinityCore/TrinityCore/issues/11859
Quest is still bugged in the 3.3.5 branch. Any chance someone could pick up where dr-j left off and fix it?

Starting at the 1:10 mark, this video shows how the quest is supposed to work. You can clearly see that after dropping the bait the wall behind Miblon disappears as he walks over to it. Miblon also does some emotes during the event. https://www.youtube.com/watch?v=-pZYz7QMluo

Rev 3761d2411968

**Comment by Rushor** [comment]
try it with sai-action-type: 99

**Comment by dr-j** [comment]
see what I writ at https://github.com/TrinityCore/TrinityCore/issues/11859 as said script does work but as usual with action 70 movement is required to see object despawn.

**Comment by tkrokli** [comment]
@dr-j : would it be possible to make the script cause the player to move, right before action 70?

**Comment by dr-j** [comment]
no as said thats the only way of making goes appear to despawn but player movement is needed to visually update object.

**Comment by tkrokli** [comment]
> but player movement is needed to visually update object

@dr-j : if you look at the YouTube video link posted in the old and closed issue #11859 (updated link with shortcut to the event: https://youtu.be/-pZYz7QMluo?t=1m9s) you can see that the wall disappears when the player moves, which should be an instinctive player reaction as if he is supposed to "give way" to Miblon Snarltooth so he can move close to the [[Bait]](http://www.wowhead.com/item=11141/bait) while he says "Grumble...". (movement script needed)
(BTW, that `creature_text` needs to be added to [Miblon Snarltooth (9467)](http://www.wowhead.com/npc=9467/miblon-snarltooth) from `broadcast_text` ID 4766).
```sql
DELETE FROM `creature_text` WHERE `entry`= 9467;
INSERT INTO `creature_text` (`entry`,`groupid`,`id`,`text`,`type`,`language`,`probability`,`emote`,`duration`,`sound`,`BroadcastTextId`,`TextRange`,`comment`) VALUES
(9467,0,0,'Grumble...',12,0,100,0,0,0,4766,0,'MIBLON_GRUMBLE');
```
(the comment label `MIBLON_GRUMBLE` is just something I thought would be a fair description)

I am pretty sure that I used to moved my player character instinctively when playing on retail back in 2010 and earlier, so I can understand why there are several comments on wowhead from users claiming that the quest was bugged. I bet those players did not move when placing the Bait, but only stood still waiting for something to happen. In my opinion it is much better to add a script that requires that the player moves to see that the wall fades/despawns than not having that option at all.

**Comment by offl** [comment]
> player movement is needed to visually update object

Is it bug?

**Comment by mario-64** [comment]
Yes. It's still bugged

**Comment by Killyana** [comment]
Related to https://github.com/TrinityCore/TrinityCore/issues/18674

**Comment by offl** [comment]
No, in this case the object disappears only when the player moves. Dunno if it is a bug or not.

**Comment by Killyana** [comment]
Yes when a gob respawn or despawn it will be updated on client side only if the player moves.

**Comment by offl** [comment]
Then the event should be implemented. That's all for this issue.

**Comment by dr-j** [comment]
well the below is what I have on this but is still hack as im sure object should not change ingame phasemask but thats how I got this to work again as previous method stopped working a while ago one thing is that update is instant by that method you dont need to move
~~~sql
UPDATE `gameobject_template` SET `AIName`='SmartGameObjectAI', `ScriptName`='' WHERE  `entry`=164758;
DELETE FROM `smart_scripts` WHERE `source_type`=1 AND `entryorguid`=164758;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(164758, 1, 0 ,0, 1, 0, 100, 1, 0, 0, 0,0,70,30,0,0,0,0,0,14,17428,164729,0,0, 0, 0, 0, 'Miblons Bait - On Spawn - Despawn Miblons Gate');
UPDATE `smart_scripts` SET `action_type`='45', `action_param1`='1', `action_param2`='1', `target_type`='19', `target_param1`='9467', `target_param2`='0', `comment`='Miblons Bait - On Spawn - Set Data on Miblon Snarltooth' WHERE  `entryorguid`=164758 AND `source_type`=1 AND `id`=0 AND `link`=0;
UPDATE `creature_template` SET `AIName`='SmartAI' WHERE  `entry`=9467;
DELETE FROM `smart_scripts` WHERE `entryorguid`=9467 AND `source_type`=0 OR  `entryorguid`=946700 AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(9467, 0, 0, 0, 38, 0, 100, 0, 1, 1, 30000, 30000, 80, 946700, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Miblon Snarltooth - on Data set - run script'),
(946700, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 48, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Miblon Snarltooth - Script - Set Active'),
(946700, 9, 1, 0, 0, 0, 100, 0, 0, 0, 0, 0, 44, 256, 0, 0, 0, 0, 0, 20, 164729, 0, 0, 0, 0, 0, 0, 'Miblon Snarltooth - Script - Change Ingame Phase of Miblons Door'),
(946700, 9, 2, 0, 0, 0, 100, 0, 30000, 30000, 0, 0, 44, 1, 0, 0, 0, 0, 0, 14, 17428, 164729	, 0, 0, 0, 0, 0, 'Miblon Snarltooth - Script - Change Ingame Phase of Miblons Door'),
(946700, 9, 3, 0, 0, 0, 100, 0, 0, 0, 0, 0, 48, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Miblon Snarltooth - Script - Set Inactive');

**Comment by offl** [comment]
Maybe SCRIPT_COMMAND_ACTIVATE_OBJECT from event_scripts will work. EventID 3718.

Actually it should be done in event_scripts(later with SAI's SMART_SCRIPT_TYPE_EVENT), I mean activating object + changing data on Miblon Snarltooth.

**Comment by offl** [comment]
Duplicate of #21910
